Configure GRUB and zipl to use the "BootLoaderSpec" boot option configuration format.

martinezjavier opened the pull request:
This pull-request contains two commits from @vathpela that adds a inst.blscfg option to configure the GRUB and zipl bootloaders to use the BootLoaderSpec (BLS) boot option configuration format.

The first commit get rids of the calls to the new-kernel-pkg script since this no longer will be available once grubby is removed from the compose.

The second commit enables the BLS configuration by default. Since this is a new option, a inst.noblscfg is also added to allow anaconda to disable its use.

These changes are part of https://fedoraproject.org/wiki/Changes/BootLoaderSpecByDefault

jkonecny12 commented on data/anaconda_options.txt:
```diff
@@ -269,3 +269,9 @@ timeout in seconds specified as a mandatory value of the option.
+Configure GRUB to use the "BootLoader Spec" boot option configuration format.
+
+noblscfg
+Configure GRUB not to use the "BootLoader Spec" boot option configuration format.
```
Why do we need to have both as boot options? Isn't one of them default behavior?

vathpela replied:
Well, you get them both anyway because that's how your option parser works, but yeah, the second description there is superfluous.

jkonecny12 commented on pyanaconda/bootloader.py:
```diff
@@ -2257,6 +2311,9 @@ def boot_dir(self):
         return "/boot"
 
     def write_config_images(self, config):
+        if flags.cmdline.getbool("blscfg", True):
```
Please do not use this string directly everywhere. The ideal solution would be to add this as bool flag to the `flags` class. However creating constant and moving it to constants would be enough.

vathpela replied:
Oh, I'd already done that and just didn't realize it would translate to flags.blscfg ; cool, will fix.

jkonecny12 commented on pyanaconda/bootloader.py:
```diff
@@ -2287,12 +2344,12 @@ def write_config_header(self, config):
                   "target=/boot\n"
-                  % {"timeout": self.timeout,
-                     "default": self.image_label(self.default)})
+                  % self.timeout)
```
We are trying to replace `%` everywhere by `.format`. Not required but would be nice :)

vathpela replied:
Okay.

vathpela commented:
I've updated it in my branch (https://github.com/vathpela/anaconda/tree/blscfg here), but Javier is on PTO until Monday, so it won't get pushed to this PR until he gets back.

martinezjavier commented:
@jkonecny12 I've updated the branch with @vathpela modifications. The changes since v1 are:
- Remove `blscfg` option.
- Check `flags.blscfg` instead of `flags.cmdline.getbool("blscfg", True)`.
- Use `.format` instead of `%` for string formatting.

jkonecny12 commented:
Thanks for the update but I'm closing this PR. 
The PR https://github.com/rhinstaller/anaconda/pull/1545 which is the same but targeted to the modularization-devel branch will propagate to master too. So this would make a duplicates.
